pkg/builtin/unary/sinh.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unary
+
+import (
+	"fmt"
+
+	"github.com/matrixorigin/matrixone/pkg/builtin"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend/overload"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/sinh"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+
+func init() {
+	extend.FunctionRegistry["sinh"] = builtin.Sinh	// register function name
+
+	for _, item := range argAndRets {
+		overload.AppendFunctionRets(builtin.Sinh, item.args, item.ret)	// append function parameter types and return types
```

Comment:
Since your sinh function returns only one type, I think this loop is quite redundant here.
perhaps this:
```go
extend.UnaryReturnTypes[builtin.Log] = func(extend extend.Extend) types.T {
		return types.T_float64
	}
```

Reply:
Fixed